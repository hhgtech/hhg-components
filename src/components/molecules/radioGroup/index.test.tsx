import 'jest-styled-components'

import React from 'react'

import { cleanup, render } from '@testing-library/react'

import { RadioGroup } from '.'
import MockTheme from '../../../misc/mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const dropdown = render(
    <MockTheme>
      <RadioGroup
        id="dropdown1"
        value="1"
        type="circle"
        isDisabled={true}
        options={[
          { label: 'option 1', value: '1' },
          { label: 'option 2', value: '2' },
          { label: 'option 3', value: '3' },
          { label: 'option 4', value: '4' },
        ]}
      />
    </MockTheme>,
  )

  const self = dropdown.container.firstChild

  expect(self).toMatchSnapshot()
})
