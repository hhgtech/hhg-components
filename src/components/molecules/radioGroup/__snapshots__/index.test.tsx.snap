// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c1[data-type='circle'] {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1[data-type='circle'] input {
  width: 20px;
  height: 20px;
  border: solid 1px #737373;
  border-radius: 100%;
  -webkit-transition: background 0.4s,border-color 0.4s,box-shadow 0.4s;
  transition: background 0.4s,border-color 0.4s,box-shadow 0.4s;
  cursor: inherit;
  margin-right: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c1[data-type='circle'] input:hover,
.c1[data-type='circle'] input:active {
  border-color: #2d87f3;
  box-shadow: 0 0 2px 2px #2d87f3;
}

.c1[data-type='circle'] input:checked {
  border-color: #2d87f3;
}

.c1[data-type='circle'] input:checked:after {
  content: '';
  position: absolute;
  width: 14px;
  height: 14px;
  left: 3px;
  top: 4px;
  border-radius: 50%;
  background-color: #2d87f3;
}

.c1[data-type='circle'] label {
  font-weight: 400;
  width: calc(100% - 24px);
  cursor: inherit;
}

.c1[data-type='circle'][data-disabled='true'] {
  color: #bfbfbf;
  cursor: not-allowed;
}

.c1[data-type='circle'][data-disabled='true'] input {
  border-color: #bfbfbf;
}

.c1[data-type='circle'][data-disabled='true'] input:hover {
  border-color: #bfbfbf;
  box-shadow: none;
}

.c1[data-type='rectangle'] input {
  display: none;
}

.c1[data-type='rectangle'] input:checked ~ label {
  font-weight: 700;
  color: #fff;
  background-color: #2d87f3;
  border-color: #2d87f3;
}

.c1[data-type='rectangle'] label {
  box-sizing: border-box;
  display: inline-block;
  width: auto;
  border: solid 1px #bfbfbf;
  border-radius: 6px;
  text-align: center;
  font-weight: 400;
  color: #262626;
  cursor: pointer;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c1[data-type='rectangle'] label:hover,
.c1[data-type='rectangle'] label:active {
  border-color: #2d87f3;
}

.c1[data-type='rectangle'][data-disabled='true'] input {
  cursor: not-allowed;
}

.c1[data-type='rectangle'][data-disabled='true'] label {
  cursor: not-allowed;
  color: #bfbfbf;
}

.c1[data-type='rectangle'][data-disabled='true'] label:hover {
  border-color: #bfbfbf;
}

.c1[data-type='rectangle'][data-size='small'] label {
  font-size: 13px;
  height: 32px;
  padding: 6px 12px;
}

.c1[data-type='rectangle'][data-size='large'] label {
  font-size: 14px;
  height: 48px;
  padding: 13px 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0[data-direction='vertical'] {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0[data-direction='vertical'] div[data-type='circle'] {
  margin-bottom: 20px;
}

.c0[data-direction='vertical'] div[data-type='rectangle'] {
  margin-bottom: 20px;
}

.c0[data-direction='horizontal'] {
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0[data-direction='horizontal'][data-group='group1'] div[data-type='rectangle'] {
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0[data-direction='horizontal'][data-group='group1'] div[data-type='rectangle']:first-child label {
  border-top-left-radius: 6px;
  border-bottom-left-radius: 6px;
}

.c0[data-direction='horizontal'][data-group='group1'] div[data-type='rectangle']:last-child label {
  border-top-right-radius: 6px;
  border-bottom-right-radius: 6px;
}

.c0[data-direction='horizontal'][data-group='group1'] div[data-type='rectangle']:not(:last-child) label {
  border-right: unset;
}

.c0[data-direction='horizontal'][data-group='group1'] div[data-type='rectangle'] label {
  border-radius: unset;
}

.c0[data-direction='horizontal'][data-group='group2'] div[data-type='rectangle'] {
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: 8px;
  margin-bottom: 8px;
}

.c0[data-direction='horizontal'] div[data-type='circle'] {
  margin-right: 25px;
}

.c0[data-direction='horizontal'] div[data-type='rectangle'] {
  margin-bottom: 20px;
}

<div
  class="c0"
  data-direction="vertical"
  data-group="group1"
>
  <div
    class="c1"
    data-disabled="true"
    data-size="small"
    data-type="circle"
  >
    <input
      checked=""
      id="dropdown1-1"
      name="dropdown1"
      type="radio"
      value="1"
    />
    <label
      for="dropdown1-1"
    >
      option 1
    </label>
  </div>
  <div
    class="c1"
    data-disabled="true"
    data-size="small"
    data-type="circle"
  >
    <input
      id="dropdown1-2"
      name="dropdown1"
      type="radio"
      value="2"
    />
    <label
      for="dropdown1-2"
    >
      option 2
    </label>
  </div>
  <div
    class="c1"
    data-disabled="true"
    data-size="small"
    data-type="circle"
  >
    <input
      id="dropdown1-3"
      name="dropdown1"
      type="radio"
      value="3"
    />
    <label
      for="dropdown1-3"
    >
      option 3
    </label>
  </div>
  <div
    class="c1"
    data-disabled="true"
    data-size="small"
    data-type="circle"
  >
    <input
      id="dropdown1-4"
      name="dropdown1"
      type="radio"
      value="4"
    />
    <label
      for="dropdown1-4"
    >
      option 4
    </label>
  </div>
</div>
`;
